code/game/objects/items/plushes.dm

```diff
@@ -542,3 +542,133 @@
+	visible_message("<span class='danger'>[src] stares at [user] angrily before going docile.</span>")
+
+/obj/item/toy/plush/goatplushie/angry/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
+	return ..()
+
+/obj/item/toy/plush/goatplushie
+	squeak_override = list('sound/items/goatsound.ogg'=1)
+
+/obj/item/toy/plush/goatplushie/angry/realgoat
+	name = "goat plushie"
+	icon_state = "realgoat"
+
+/obj/item/toy/plush/realgoat
+	name = "goat plushie"
+	desc = "Despite its cuddly appearance and plush nature, it will beat you up all the same, or atleast it would if it wasnt a normal plushie."
+	icon_state = "realgoat"
+	squeak_override = list('sound/items/goatsound.ogg'=1)
+
+/obj/item/toy/plush/goatplushie/angry/kinggoat
+	name = "King Goat Plushie"
+	desc = "A plushie depicting the king of all goats."
+	icon_state = "kinggoat"
+	cooldown_modifier = 3
+	throwforce = 25
+	force = 25
+	attack_verb = list("chomped")
+
+/obj/item/toy/plush/goatplushie/angry/ascendedkinggoat
+	name = "Ascended King Goat Plushie"
+	desc = "A plushie depicting the god of all goats."
+	icon_state = "ascendedkinggoat"
+	cooldown_modifier = 1
+	throwforce = 30
+	force = 30
+	attack_verb = list("chomped")
+
+/obj/item/toy/plush/goatplushie/angry/ascendedkinggoat/attackby(obj/item/I,mob/living/user,params)
+	if(I.get_sharpness())
+		user.visible_message("<span class='notice'>[user] attempts to stab [src]!</span>", "<span class='suicide'>The knife bounces off [src]'s back before breaking into millions of pieces... [src] glares at [user]!</span>") // You fucked up now son
+		I.play_tool_sound(src)
+		qdel(I)
+		user.gib()
+
+/obj/item/toy/plush/goatplushie/angry/kinggoat/attackby(obj/item/I,mob/living/user,params)
+	if(I.get_sharpness())
+		user.visible_message("<span class='notice'>[user] stabs [src] to shreds!</span>", "<span class='notice'>[src]'s death has attracted the attention of the king goat plushie guards!</span>")
+		I.play_tool_sound(src)
+		qdel(src)
+		var/turf/location = get_turf(user)
+		new/obj/item/toy/plush/goatplushie/angry/masterguardgoat(location)
+		new/obj/item/toy/plush/goatplushie/angry/masterguardgoat(location)
+		new/obj/item/toy/plush/goatplushie/angry/masterguardgoat(location)
+		new/obj/item/toy/plush/goatplushie/angry/masterguardgoat(location)
+		new/obj/item/toy/plush/goatplushie/angry/guardgoat(location)
+		new/obj/item/toy/plush/goatplushie/angry/guardgoat(location)
+		new/obj/item/toy/plush/goatplushie/angry/guardgoat(location)
+		new/obj/item/toy/plush/goatplushie/angry/guardgoat(location)
+		new/obj/item/toy/plush/goatplushie/angry/guardgoat(location)
+		new/obj/item/toy/plush/goatplushie/angry/guardgoat(location)
+		new/obj/item/toy/plush/goatplushie/angry/guardgoat(location)
+		new/obj/item/toy/plush/goatplushie/angry/guardgoat(location)
+
+/obj/item/toy/plush/goatplushie/angry/kinggoat/attackby(/obj/item/reagent_containers/food/snacks/grown/cabbage/C,mob/living/user,params)
+		var/C = /obj/item/reagent_containers/food/snacks/grown/cabbage
```

Comment:
This should be an istype check, not a separate proc. Maybe byond actually does overloading, but i'm pretty sure it's out of the codebase's standard.